command/agent/agent_endpoint_test.go

```diff
@@ -386,6 +386,46 @@ func TestHTTP_AgentMonitor(t *testing.T) {
 				require.Fail(t, err.Error())
 			})
 		}
+
+		// stream logs for a local client
+		{
+			req, err := http.NewRequest("GET", "/v1/agent/monitor?log_level=warn", nil)
+			require.Nil(t, err)
+			resp := newClosableRecorder()
+			defer resp.Close()
+
+			go func() {
+				// set server to nil to monitor as client
+				s.Agent.server = nil
+				_, err = s.Server.AgentMonitor(resp, req)
+				require.NoError(t, err)
```

Comment:
I would suggest using separate tests or `Run()` invocations.  Given that this mutates the agent, it may interfere unexpectedly with other tests, if this section is moved or new tests are added.  Also, `require.NoError` doesn't work so well in a goroutine, as it uses a panic; you can use `assert.NoError` here or require the error in the main test goroutine.

Reply:
:+1: replaced closures with Run()